forge-gui/res/cardsfolder/upcoming/break_out.txt

```diff
@@ -0,0 +1,11 @@
+Name:Break Out
+ManaCost:R G
+Types:Sorcery
+A:SP$ PeekAndReveal | Defined$ You | PeekAmount$ 6 | NoReveal$ True | RememberPeeked$ True | SubAbility$ PickOne | StackDescription$ SpellDescription | SpellDescription$ Look at the top six cards of your library. You may reveal a creature card from among them. If that card has mana value 2 or less, you may put it onto the battlefield and it gains haste until end of turn. If you didn’t put the revealed card onto the battlefield this way, put it into your hand. Put the rest on the bottom of your library in a random order.
```

Comment:
why don't you use RevealValid + RevealOptional + RememberRevaled = no extra ChooseCard?

Reply:
I tried a good number of iterations on those but it frankly got me nowhere as reasonable.
I reached the solution above after looking over the few examples of `PeekAndReveal` used for similar effects (I borrowed heavily from Nine-Fingers Keene's triggered ability) and noticing there's a `SubAbility` for the "put the rest on the bottom of your library in a random order" part of the effect. For `Dig` that's built-in setting. I suspected there could be a conflict for `PeekAndReveal` when using `RememberRevealed` since the revealed card and the cards that are put on the bottom of the library are called back to by `Remembered` (or `IsRemembered`). I'm certainly not disabused of that notion.